src/config.ts

```diff
@@ -1,361 +1,153 @@
-// Sketch
-//
-// Need an easy way of getting and setting settings
-// If a setting is not set, the default should probably be returned.
-// That probably means that binds etc. should be per-key?
-//
-// We should probably store all settings in memory, and only load from storage on startup and when we set it
-//
-// Really, we'd like a way of just letting things use the variables
-//
+import DEFAULTS from './config_defaults'
+import * as Utils from './utils'
 
-const CONFIGNAME = "userconfig"
-const WAITERS = []
-let INITIALISED = false
+export const CONFIG_NAME = 'config'
+const asyncGetters: (() => void)[] = []
+let initialised = false
```

Comment:
Are GLOBAL_VARIABLES in all caps frowned on in JS?

Reply:
Generally, ALL_CAPS are for CONSTANTS. At least that's the convention I've always used. That's not a global variable; it's a top-level variable in the module.